common/os_calls.c

```diff
@@ -2218,6 +2218,14 @@ g_file_open_ex(const char *file_name, int aread, int awrite,
 #endif
 }
 
+/*****************************************************************************/
+/* returns -1 on error, else return handle or file descriptor */
+int
+g_file_open_ro(const char *file_name)
```

Comment:
👍  I like this idea.

Reply:
I do too. I'm also quite happy with the idea of setting up a `g_file_open_rw()` as the special case as you hint at above, @iskunk. If you want to put the extra work in to do it that way, that's fine by me.